internal/database/drivers/mongodb/mongodb.go

```diff
@@ -154,25 +154,35 @@ func (m *Database) collections(ctx context.Context, filter bson.D) ([]string, er
 }
 
 // GetEvents gets all events information.
-func (m *Database) GetEvents(ctx context.Context, request requests.MultipleEventsRequest) ([]drivers.EiffelEvent, error) {
+func (m *Database) GetEvents(ctx context.Context, request requests.MultipleEventsRequest) ([]drivers.EiffelEvent, int64, error) {
 	filter, err := buildFilter(request.Conditions)
 	if err != nil {
 		m.logger.Errorf("Database: %v", err)
-		return nil, err
+		return nil, 0, err
 	}
 	collections, err := m.collections(ctx, filter)
 	if err != nil {
 		m.logger.Errorf("Database: %v", err)
-		return nil, err
+		return nil, 0, err
 	}
 
 	m.logger.Debugf("fetching events from %d collections", len(collections))
 	var allEvents []drivers.EiffelEvent
+	var numberOfDocuments int64
 	for _, collection := range collections {
 		var events []drivers.EiffelEvent
-		cursor, err := m.database.Collection(collection).Find(ctx, filter,
-			// Remove the _id field from the resulting document.
-			options.Find().SetProjection(bson.M{"_id": 0}))
+		findOptions := options.Find()
+		// Remove the _id field from the resulting document.
+		findOptions.SetProjection(bson.M{"_id": 0})
+		// PageNo starts at 1 instead of 0 so we need to decrement it by 1 here.
+		findOptions.SetSkip(int64((request.PageNo - 1) * request.PageSize))
+		findOptions.SetLimit(int64(request.PageSize))
```

Comment:
Optional: The builder pattern can be used to reduce repetition:

```suggestion
		findOptions := options.Find()
			.SetProjection(bson.M{"_id": 0}) // Remove the _id field from the resulting document.
			.SetSkip(int64((request.PageNo - 1) * request.PageSize)) // PageNo starts at 1 instead of 0 so we need to decrement it by 1 here.
			.SetLimit(int64(request.PageSize))
```

Reply:
Done